### PR TITLE
specific error in test for mint module

### PR DESCRIPTION
**Pull request by ThanhNhann**
Closes: #2959

## What is the purpose of the change

Update tests that are using expectPass or expectErr as the bool variable to the error type in the mint module

## Brief Changelog

- Coverage mint module
- All tests pass
- Commented at test cases where the error returned is not as expected

### CODE REVIEW

File: x/mint/keeper/keeper_test.go [hunk 1]
```diff
@@ -403,8 +403,9 @@ func (suite *KeeperTestSuite) TestDistributeToModule() {
 				actualMintModuleBalanceAmount := bankKeeper.GetBalance(ctx, accountKeeper.GetModuleAddress(types.ModuleName), tc.mintedCoin.Denom).Amount
 				actualRecepientModuleBalanceAmount := bankKeeper.GetBalance(ctx, accountKeeper.GetModuleAddress(tc.recepientModule), tc.mintedCoin.Denom).Amount
 
-				if tc.expectedError {
+				if tc.expectedError != nil {
 					suite.Require().Error(err)
+					suite.Require().Equal(tc.expectedError.Error(), err.Error())
```
Comment (p0mvn) [inline x/mint/keeper/keeper_test.go:408]: Any reason for not using `errors.As` here?

File: x/mint/keeper/hooks_test.go [hunk 2]
```diff
@@ -393,14 +394,17 @@ func (suite *KeeperTestSuite) TestAfterEpochEnd() {
 			oldSupply := app.BankKeeper.GetSupply(ctx, sdk.DefaultBondDenom).Amount
 			suite.Require().Equal(sdk.NewInt(keeper.DeveloperVestingAmount), oldSupply)
 
-			if tc.expectedError {
-				suite.Require().Error(mintKeeper.AfterEpochEnd(ctx, defaultEpochIdentifier, tc.hookArgEpochNum))
+			err := mintKeeper.AfterEpochEnd(ctx, defaultEpochIdentifier, tc.hookArgEpochNum)
+			if tc.expectedError != nil {
+				suite.Require().Error(err)
+				suite.Require().ErrorAs(err, &tc.expectedError)
+				println(tc.expectedError.Error())
```
Comment (p0mvn) [inline x/mint/keeper/hooks_test.go:401]:
```suggestion
```

File: x/mint/keeper/hooks_test.go [hunk 3]
```diff
@@ -393,14 +394,17 @@ func (suite *KeeperTestSuite) TestAfterEpochEnd() {
 			oldSupply := app.BankKeeper.GetSupply(ctx, sdk.DefaultBondDenom).Amount
 			suite.Require().Equal(sdk.NewInt(keeper.DeveloperVestingAmount), oldSupply)
 
-			if tc.expectedError {
-				suite.Require().Error(mintKeeper.AfterEpochEnd(ctx, defaultEpochIdentifier, tc.hookArgEpochNum))
+			err := mintKeeper.AfterEpochEnd(ctx, defaultEpochIdentifier, tc.hookArgEpochNum)
```
Comment (p0mvn) [inline x/mint/keeper/hooks_test.go:397]: nit: I think missing a tab here

File: x/mint/keeper/keeper_test.go (hunk 1 above)
Comment (ThanhNhann) [reply x/mint/keeper/keeper_test.go:408]: With this test, the error is detail, so I think `Equal` is more suitable

File: x/mint/keeper/hooks_test.go (hunk 3 above)
Comment (ThanhNhann) [reply x/mint/keeper/hooks_test.go:397]: Maybe not sir, I updated the code with a new line but it don't change any logic

**Comment by czarcas7ic** [comment]
Hey @ThanhNhann thanks for the PR! There was some internal confusion on what was discussed on desired endstate for tests. Going to close this PR for now and we can potentially reopen after further discussions

**Pull request closed without merging**